pycodestyle.py

```diff
@@ -1521,6 +1526,10 @@ def ambiguous_identifier(logical_line, tokens):
         # find function definitions
         if prev_text == 'def':
             is_func_def = True
+        if prev_text == 'lambda':
+            is_lambda_arg = True
+        if is_lambda_arg and prev_text == ":":
+            is_lambda_arg = False
```

Comment:
this _technically_ isn't precise enough due to lambda default arguments:

```pycon
>>> x = [1, 2, 3]
>>> y = lambda a=x[1:]: a
>>> y()
[2, 3]
```

Reply:
good callout! how bout now?

Reply:
Also, along these lines there's a false positive in pycodestyle now of:

```python
def a(x=l[1:5]):
    ...
```

i fixed it for lambdas in this diff, but we could make an issue / fix it for function defs after?

Reply:
https://github.com/PyCQA/pycodestyle/issues/1114